### PR TITLE
Issue 211 - Check lambda functions for number of arguments

### LINKED ISSUE #211: Feature: check `lambda` functions for the number of arguments
We already have the logic to check regular an `async` functions for the number of arguments.
https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/visitors/ast/complexity/function.py#L155

But we need to check `lambda` functions as well. Since one can create a `lambda` function that receives too many arguments. 

What needs to be done:
1. We need to implement a new handler for `Lambda` node in the visitor above
2. We need to unit test this behavior here: https://github.com/wemake-services/wemake-python-styleguide/blob/master/tests/test_checkers/test_high_complexity.py

### PR DESCRIPTION
**Pull request by tommbee**
# I have made things!

Add lambda function support for async functions (Closes #211 )

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #211

### CODE REVIEW

File: wemake_python_styleguide/visitors/ast/complexity/function.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 from wemake_python_styleguide.visitors.base import BaseNodeVisitor
 from wemake_python_styleguide.visitors.decorators import alias
 
-FunctionCounter = DefaultDict[AnyFunctionDef, int]
+FunctionCounter = DefaultDict[Union[AnyFunctionDef, ast.Lambda], int]
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/complexity/function.py]: Let's crate a new type alias `AnyFunctionAndLambda = Union[AnyFunctionDef, ast.Lambda]`

File: wemake_python_styleguide/visitors/ast/complexity/function.py [hunk 2]
```diff
@@ -82,6 +82,18 @@ def check_arguments_count(self, node: AnyFunctionDef) -> None:
 
         self.arguments[node] = counter - has_extra_arg
 
+    def check_lambda_arguments_count(self, node: ast.Lambda) -> None:
+        """Checks the number of the arguments in a lambda function."""
+        counter = 0
+
+        counter += len(node.args.args) + len(node.args.kwonlyargs)
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/complexity/function.py]: This code should be reused from regular functions check. It is 100% similar.

File: wemake_python_styleguide/visitors/ast/complexity/function.py [hunk 3]
```diff
@@ -124,14 +136,16 @@ def _check_possible_switch(self) -> None:
     def _check_function_internals(self) -> None:
         for node, variables in self._counter.variables.items():
             if len(variables) > self.options.max_local_variables:
+                text = node.name if not isinstance(node, ast.Lambda) else None
                 self.add_violation(
-                    TooManyLocalsViolation(node, text=node.name),
+                    TooManyLocalsViolation(node, text=text),
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/complexity/function.py]: There can not be any local variables in a `lambda`.

File: wemake_python_styleguide/visitors/ast/complexity/function.py [hunk 4]
```diff
@@ -143,8 +157,9 @@ def _check_function_signature(self) -> None:
 
         for node, returns in self._counter.returns.items():
             if returns > self.options.max_returns:
+                text = node.name if not isinstance(node, ast.Lambda) else None
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/complexity/function.py]: There can not be any `return` statements in a `lambda`.

File: wemake_python_styleguide/visitors/ast/complexity/function.py [hunk 5]
```diff
@@ -167,3 +182,18 @@ def visit_any_function(self, node: AnyFunctionDef) -> None:
         self._counter.check_arguments_count(node)
         self._counter.check_function_complexity(node)
         self.generic_visit(node)
+
+    def visit_Lambda(self, node: ast.Lambda) -> None:
+        """
+        Checks lambda function's internal complexity.
+
+        Raises:
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/complexity/function.py]: Please, check what errors can be raised by this visitor. I guess it is only a single one: arguments count.

**Comment by sobolevn** [comment]
Big thank. Much wow!

**Comment by tommbee** [comment]
🙌 Thanks for your patience, still getting my head around the codebase.

**Pull request merged**